core-processor/src/processing.rs

```diff
@@ -139,25 +141,60 @@ where
     });
 
     match exec_result {
-        Ok(res) => Ok(match res.kind {
-            DispatchResultKind::Trap(reason) => process_execution_error(
-                res.dispatch,
-                program_id,
-                res.gas_amount.burned(),
-                res.system_reservation_context,
-                ActorExecutionErrorReplyReason::Trap(reason),
-            ),
-            DispatchResultKind::Success => process_success(Success, res),
-            DispatchResultKind::Wait(duration, ref waited_type) => {
-                process_success(Wait(duration, waited_type.clone()), res)
-            }
-            DispatchResultKind::Exit(value_destination) => {
-                process_success(Exit(value_destination), res)
-            }
-            DispatchResultKind::GasAllowanceExceed => {
-                process_allowance_exceed(dispatch, program_id, res.gas_amount.burned())
+        Ok(res) => {
+            match res.kind {
+                DispatchResultKind::Success
+                | DispatchResultKind::Wait(_, _)
+                | DispatchResultKind::Exit(_) => {
+                    // assert that after processing the initial reservation is less or equal to the current one.
+                    // during execution reservation amount might increase due to `system_reserve_gas` calls
+                    // thus making initial reservation less than current one.
+                    debug_assert!(system_reservation_ctx
+                        .current_reservation
```

Comment:
it's always `None`, why check it?

Reply:
still if you want to check that, you must use `unwrap_or_default` for the `system_reservation_ctx.current_reservation` (the one created before execution)